packages/syft/src/syft/service/worker/worker_pool_service.py

```diff
@@ -178,6 +179,109 @@ def delete_worker(
+        result = self.stash.get_by_name(context.credentials, pool_name=pool_name)
+        if result.is_err():
+            return SyftError(message=f"{result.err()}")
+        if result.ok() is None:
+            return SyftError(
+                message=f"Worker Pool with name: {pool_name} was not found"
+            )
+
+        worker_pool = result.ok()
+
+        worker = None
+        for w in worker_pool.workers:
+            if w.id == worker_id:
+                worker = w
+                break
+
+        if worker is None:
+            return SyftError(
+                message=f"Worker with id: {worker_id} not found in pool: {worker_pool.name}"
+            )
+
+        worker_status = self.get_worker_status(
+            context=context, pool_name=pool_name, worker_id=worker.id
+        )
+        if isinstance(worker_status, SyftError):
+            return worker_status
+        elif worker_status != WorkerStatus.PENDING:
+            for worker in worker_pool.workers:
+                if worker.id == worker_id:
+                    worker.status = worker_status
+                    break
+            result = self.stash.update(
+                credentials=context.credentials,
+                obj=worker_pool,
+            )
+            if result.is_err():
+                return SyftError(message=f"{result.err()}")
+            else:
+                return worker
+        else:
+            return worker
+
+    @service_method(
+        path="worker_pool.get_worker_status",
+        name="get_worker_status",
+        roles=DATA_OWNER_ROLE_LEVEL,
+    )
+    def get_worker_status(
+        self, context: AuthedServiceContext, pool_name: str, worker_id: UID
+    ) -> Union[WorkerStatus, SyftError]:
+        result = self.stash.get_by_name(context.credentials, pool_name=pool_name)
+        if result.is_err():
+            return SyftError(message=f"{result.err()}")
+
+        if result.ok() is None:
+            return SyftError(
+                message=f"Worker Pool with name: {pool_name} was not found"
+            )
+
+        worker_pool = result.ok()
+
+        worker = None
+        for w in worker_pool.workers:
+            if w.id == worker_id:
+                worker = w
+                break
+
+        if worker is None:
+            return SyftError(
+                message=f"Worker with id: {worker_id} not found in pool: {worker_pool.name}"
+            )
+        try:
+            client = docker.from_env()
+            worker_status = client.containers.get(worker.container_id).status
+            if worker_status == "running":
+                worker.status = WorkerStatus.RUNNING
+            elif worker_status in ["paused", "removing", "exited", "dead"]:
+                worker.status = WorkerStatus.STOPPED
+            elif worker_status == "restarting":
+                worker.status = WorkerStatus.RESTARTED
+            elif worker_status == "created":
+                worker.status = WorkerStatus.PENDING
```

Comment:
MAYBE: Since we update the worker status and as and when we call the method get_worker, we could modify this dependency to be something like

```python
def get_worker():
fetch from store
update the status
save to db
return the update worker

```

```python3
def get_worker_status(args, kwargs)
return self.get_worker( args, kwargs).status
```
Thoughts?

Reply:
I like it, this way the status is saved to db for both calls. I'll look at making the changes when I get a chance on Sunday.

Reply:
Changes made to latest commit.